src/components/plugin-selector-dialog.js

```diff
@@ -0,0 +1,57 @@
+import PropTypes from 'prop-types';
+import { Modal } from 'antd';
+import React, { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import InfoFactory from '../plugins/info-factory.js';
+import { useService } from './container-context.js';
+import GridSelector from './grid-selector.js';
+
+function PluginSelectorDialog({ visible, onSelected, onCancel }) {
+  const { t } = useTranslation('pluginSelectorDialog');
+  const infoFactory = useService(InfoFactory);
+
+  const pluginItems = useMemo(() => {
+    return infoFactory.getRegisteredTypes()
+      .map(typeName => infoFactory.createInfo(typeName))
+      .map(info => ({
+        key: info.type,
+        icon: info.getIcon?.(t) || null,
+        label: info.getName(t)
+      }));
+  }, [infoFactory, t]);
+
+  const [selectedPluginType, setSelectedPluginType] = useState(pluginItems[0]?.key || null);
+
+  const handleOk = () => {
+    onSelected?.(selectedPluginType);
+  };
+
+  const handleCancel = () => onCancel?.();
```

Comment:
Should we reset `selectedPluginType` on cancel (or might need to be on ok), so that the first item is always selected when re-opening the modal? 🤔

Reply:
Actually, I intended to persist the last choice in session storage until I realized that it's working out of the box in the React state. I think it might be logical to the user to pre-select the last selected plugin while being on the same page.

Reply:
So it was intentional 👍  Ok, then let's have it preselected.